packages/flutter/lib/src/material/navigation_drawer.dart

```diff
@@ -0,0 +1,416 @@
+    final NavigationDrawerThemeData defaults =
+        _NavigationDrawerDefaultsM3(context);
+
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: 12.0),
+      child: NavigationDestinationSemantics(
+        child: SizedBox(
+          height: navigationDrawerTheme.tileHeight ?? defaults.tileHeight,
+          child: InkWell(
+            highlightColor: Colors.transparent,
+            onTap: info.onTap,
+            borderRadius: const BorderRadius.all(Radius.circular(28.0)),
+            child: Stack(
+              alignment: Alignment.center,
+              children: <Widget>[
+                NavigationIndicator(
+                  animation:
+                      NavigationDestinationInfo.of(context).selectedAnimation,
+                  color: navigationDrawerTheme.indicatorColor ??
+                      defaults.indicatorColor!,
+                  shape: navigationDrawerTheme.indicatorShape ??
+                      defaults.indicatorShape!,
+                  width: (navigationDrawerTheme.indicatorSize ??
+                          defaults.indicatorSize!)
+                      .width,
+                  height: (navigationDrawerTheme.indicatorSize ??
+                          defaults.indicatorSize!)
+                      .height,
+                ),
+                Row(
+                  children: <Widget>[
+                    const SizedBox(width: 16),
+                    buildIcon(context),
+                    const SizedBox(width: 12),
+                    buildLabel(context),
+                  ],
+                ),
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+/// Returns `true` if this animation is ticking forward, or has completed,
+/// based on [status].
+bool _isForwardOrCompleted(Animation<double> animation) {
+  return animation.status == AnimationStatus.forward ||
+      animation.status == AnimationStatus.completed;
+}
+
+// BEGIN GENERATED TOKEN PROPERTIES - NavigationDrawer
+
+// Do not edit by hand. The code between the "BEGIN GENERATED" and
+// "END GENERATED" comments are generated from data in the Material
+// Design token database by the script:
+//   dev/tools/gen_defaults/bin/gen_defaults.dart.
+
+// Token database version: v0_101
+
+class _NavigationDrawerDefaultsM3 extends NavigationDrawerThemeData {
+  const _NavigationDrawerDefaultsM3(this.context)
+      : super(
+          elevation: 1.0,
+        );
+
+  final BuildContext context;
+
+  @override
+  double? get tileHeight => 56.0;
+
+  @override
+  Color? get backgroundColor => Theme.of(context).colorScheme.surface;
+
+  @override
+  Color? get surfaceTintColor => Theme.of(context).colorScheme.surfaceTint;
+
+  @override
+  Color? get indicatorColor => Theme.of(context).colorScheme.secondaryContainer;
+
+  @override
+  ShapeBorder? get indicatorShape => const StadiumBorder();
+
+  @override
+  Size? get indicatorSize => const Size(336.0, 56.0);
```

Comment:
Same for these. They could go in the constructor.

Reply:
Done